kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/MctsBeneficiaryImportServiceImpl.java

```diff
@@ -0,0 +1,490 @@
+        DateTime lmp = (DateTime) record.get(LMP);
+
+        String errors = setLocationFields(record, mother);
+        if (errors != null) {
+            rejectBeneficiary(msisdn, SubscriptionRejectionReason.INVALID_LOCATION, SubscriptionPackType.PREGNANCY);
+            return;
+        }
+
+        if (lmp == null) {
+            rejectBeneficiary(msisdn, SubscriptionRejectionReason.MISSING_LMP, SubscriptionPackType.PREGNANCY);
+            return;
+        }
+
+        // TODO: more data validation specified in #111
+        mother.setName(name);
+
+        processSubscriptionForBeneficiary(mother, msisdn, lmp, SubscriptionPackType.PREGNANCY);
+    }
+
+
+    private void importChildRecord(Map<String, Object> record) {
+        MctsChild child = (MctsChild) record.get(BENEFICIARY_ID);
+        String name = (String) record.get(BENEFICIARY_NAME);
+        Long msisdn = (Long) record.get(MSISDN);
+        MctsMother mother = (MctsMother) record.get(MOTHER_ID);
+        DateTime dob = (DateTime) record.get(DOB);
+
+        String errors = setLocationFields(record, child);
+        if (errors != null) {
+            rejectBeneficiary(msisdn, SubscriptionRejectionReason.INVALID_LOCATION, SubscriptionPackType.CHILD);
+            return;
+        }
+
+        if (dob == null) {
+            rejectBeneficiary(msisdn, SubscriptionRejectionReason.MISSING_DOB, SubscriptionPackType.CHILD);
+            return;
+        }
+        // TODO: more data validation specified in #111
+
+        child.setName(name);
+        child.setMother(mother);
+
+        processSubscriptionForBeneficiary(child, msisdn, dob, SubscriptionPackType.CHILD);
+    }
+
+
+    private void processSubscriptionForBeneficiary(MctsBeneficiary beneficiary, Long msisdn, DateTime referenceDate,
+                                                  SubscriptionPackType type) {
+        Language language = beneficiary.getDistrict().getLanguage();
+        SubscriptionPack pack = subscriptionPackDataService.byType(type);
+        Subscriber subscriber = subscriberService.getSubscriber(msisdn);
+
+        if (subscriber == null) {
+            // there's no subscriber with this MSISDN, create one
+
+            subscriber = new Subscriber(msisdn, language);
+            subscriber = updateSubscriber(subscriber, beneficiary, referenceDate, type);
+            subscriberService.create(subscriber);
+
+        } else if (subscriptionService.subscriberHasActiveSubscription(subscriber, type)) {
+            // subscriber already has an active subscription to this pack
+
+            MctsBeneficiary existingBeneficiary = (type == SubscriptionPackType.PREGNANCY) ? subscriber.getMother() :
+                    subscriber.getChild();
+
+            if (existingBeneficiary == null) {
+                // there's already an IVR-originated subscription for this MSISDN
+                rejectBeneficiary(msisdn, SubscriptionRejectionReason.ALREADY_SUBSCRIBED, type);
```

Comment:
is there no requirement to tell this rejection apart from the following else if rejection reason?